src/pudl/metadata/classes.py

```diff
@@ -1789,9 +1789,11 @@ def enforce_schema(self, df: pd.DataFrame) -> pd.DataFrame:
 
         df = self.format_df(df)
         pk = self.schema.primary_key
-        if pk and not (dupes := df[df.duplicated(subset=pk)]).empty:
+        if pk and not (dupes := df[df.duplicated(subset=pk, keep=False)]).empty:
```

Comment:
I set keep=False here because it's more informative to see all of the duplicate records when the PKs are not unique -- or at least a sampling of them, when there are too many dupes to display.